modules/custom/az_migration/src/Plugin/migrate/process/ParagraphsMapFieldPropertyValue.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Drupal\az_migration\Plugin\migrate\process;
+
+use Drupal\migrate\MigrateExecutableInterface;
+use Drupal\migrate\ProcessPluginBase;
+use Drupal\migrate\Row;
+
+/**
+ * Process Plugin to Field map property value for paragraphs.
+ *
+ * @MigrateProcessPlugin(
+ *   id = "paragraphs_field_property_mapping"
+ * )
+ */
+class ParagraphsMapFieldPropertyValue extends ProcessPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function transform($value, MigrateExecutableInterface $migrate_executable, Row $row, $destination_property) {
+    // Getting the field property values from the field collection.
+    $key = $this->configuration['key'];
+    $value_key = $this->configuration['value'];
+    // Get the field values.
+    if ($key === 'body_format') {
+      $value[$key] = $value_key;
+    }
+    else {
+      $field_name = $this->configuration['field_name'];
+      $field_values = $row->getSourceProperty($field_name);
+      $value[$key] = $field_values[$value['delta']][$value_key];
+    }
+    return $value;
```

Comment:
is this supposed to be `$value[$key]` rather than `$value`?

Reply:
No @tadean , it should be value. We are mapping the value to the key and returning the entire value to migrate.